w11b01/test/pgdp/pengusurvivors/GenerateModelTest.java

```diff
@@ -76,6 +76,71 @@ public void fullyConnected3on3() {
         assertEquals(3, maxPairs);
     }
 
+    @Test
+    public void emptyWorkaholics() {
+        int[] workaholics = {};
+        int[] procrastinators = {1, 2, 3};
+        int[][] friendships = {};
+        FlowGraph graph = PenguSurvivors.generateModel(workaholics, procrastinators, friendships);
+        int maxPairs = graph.computeMaxFlowValue();
+        assertEquals(0, maxPairs);
+    }
+
+    @Test
+    public void emptyProcrastinators() {
+        int[] workaholics = {1, 2, 3};
+        int[] procrastinators = {};
+        int[][] friendships = {};
+        FlowGraph graph = PenguSurvivors.generateModel(workaholics, procrastinators, friendships);
+        int maxPairs = graph.computeMaxFlowValue();
+        assertEquals(0, maxPairs);
+    }
+
+    @Test
+    public void nullWorkaholics() {
+        int[] workaholics = null;
+        int[] procrastinators = {1, 2, 3};
+        int[][] friendships = {};
+        FlowGraph graph = PenguSurvivors.generateModel(workaholics, procrastinators, friendships);
+        if (graph != null) {
+            int maxPairs = graph.computeMaxFlowValue();
+            assertEquals(0, maxPairs);
+        }
+    }
+
+    @Test
+    public void nullProcrastinators() {
+        int[] workaholics = {1, 2, 3};
+        int[] procrastinators = null;
+        int[][] friendships = {};
+        FlowGraph graph = PenguSurvivors.generateModel(workaholics, procrastinators, friendships);
+        if (graph != null) {
+            int maxPairs = graph.computeMaxFlowValue();
+            assertEquals(0, maxPairs);
+        }
+    }
+
+    @Test
+    public void nullFriendships() {
+        int[] workaholics = {1, 2, 3};
+        int[] procrastinators = {1, 2, 3};
+        int[][] friendships = null;
+        FlowGraph graph = PenguSurvivors.generateModel(workaholics, procrastinators, friendships);
+        if (graph != null) {
+            int maxPairs = graph.computeMaxFlowValue();
+            assertEquals(0, maxPairs);
+        }
+    }
+
+    @Test
+    public void friendshipOutOfBounds() {
```

Comment:
The Artemis tests only give valid inputs. The edge cases aren't tested there anyways.